packages/notebook/src/actions.ts

```diff
@@ -1097,11 +1097,14 @@ namespace Private {
   function cloneCell(model: INotebookModel, cell: ICellModel): ICellModel {
     switch (cell.type) {
     case 'code':
-      return model.contentFactory.createCodeCell(cell.toJSON());
+      // TODO why isnt modeldb or id passed here?
+      return model.contentFactory.createCodeCell({ cell: cell.toJSON() });
```

Comment:
this factory api is awkward, it accepts full cell options, but ignores things like modeldb.